engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -870,6 +873,21 @@ private void execute() throws YarnException, IOException
       }
 
       numRequestedContainers += containerRequests.size() - removedContainerRequests.size();
+
```

Comment:
Regarding 
1. sendContainerAskToRM is already overloaded, as kill request is sent to NM and not to RM, so it will only complicate it further.
2. We can't move this logic to NMCallbackHandler. This PR is because onContainerStopped may not have been called.